deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +386,190 @@ public boolean cleanUp() {
+					/*pod initialization activity completed*/
+					if (initializedPods.size() == replicas && !initializationActivityDone) {
+						initializationActivityDone = true;
+						WorkflowLogger.endActivity(initializedActivityContext, Status.DONE);
+					}
+				}
+
+				if (initializationActivityDone && !creationActivityStarted) {
+					activityContext=creationActivityContext;
+					WorkflowLogger.startActivity(creationActivityContext);
+					creationActivityStarted = true;
+				}
+				if (PodPredicates.isPodCreated().test(item.object)) {
+					createdPods.add(item.object.getMetadata().getName());
+					/*pod creation activity completed*/
+					if (createdPods.size() == replicas && !creationActivityDone) {
+						WorkflowLogger.endActivity(creationActivityContext, Status.DONE);
+						creationActivityDone = true;
+					}
+				}
+
+				if (creationActivityDone && !readyActivityStarted) {
+					activityContext=readyActivityContext;
+					WorkflowLogger.startActivity(readyActivityContext);
+					readyActivityStarted = true;
+				}
+				if (PodPredicates.isPodReady().test(item.object)) {
+					readyPods.add(item.object.getMetadata().getName());
+					/*pod readiness activity completed*/
+					if (readyPods.size() == replicas) {
+						WorkflowLogger.endActivity(readyActivityContext, Status.DONE);
+						break;
+					}
+				}
+
+				if (readyPods.size() == replicas) {
+					break;
+				}
+
+			}
+			
+			if(!initializationActivityDone) {
+				WorkflowLogger.endActivity(initializedActivityContext, Status.FAILED);
+				throw new HyscaleException(DeployerErrorCodes.FAILED_TO_INITIALIZE_POD);
+			}
+			if(!creationActivityDone) {
+				WorkflowLogger.endActivity(creationActivityContext, Status.FAILED);
+				throw new HyscaleException(DeployerErrorCodes.FAILED_TO_CREATE_POD);
+			}
+			if(readyPods.size()!=replicas) {
+				WorkflowLogger.endActivity(readyActivityContext, Status.FAILED);
+				throw new HyscaleException(DeployerErrorCodes.POD_FAILED_READINESS);
+			}
+			
+			
+		} finally {
+			try {
+				watch.close();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+
+	/**
+	 * It will read object(v1Deployment or v1StatefulSet) from cluster and return along with their resource kind
+	 * 
+	 * @param apiClient
+	 * @param appName
+	 * @param namespace
+	 * @return podParent
+	 */
+	private PodParent getPodParent(ApiClient apiClient, String serviceName, String namespace) throws HyscaleException{ 
+		V1StatefulSet v1StatefulSet = null;
+		V1Deployment v1Deployment = null;
+		PodParent podParent=new PodParent();
+		try {
+			V1StatefulSetHandler v1StatefulSetHandler = (V1StatefulSetHandler) ResourceHandlers
+					.getHandlerOf(ResourceKind.STATEFUL_SET.getKind());
+			v1StatefulSet = v1StatefulSetHandler.get(apiClient, serviceName, namespace);
+			if (v1StatefulSet != null) {
+				podParent.setKind(ResourceKind.STATEFUL_SET.getKind());
+				podParent.setParent(v1StatefulSet);
+			}
```

Comment:
Here we can return podParent. Then no need to have a if condition below
Fix indentation in function